Userland/Libraries/LibWeb/HTML/HTMLImageElement.cpp

```diff
@@ -218,4 +219,536 @@ Optional<ARIA::Role> HTMLImageElement::default_role() const
+
+        // 20. Set request's referrer policy to the current state of the element's referrerpolicy attribute.
+        request->set_referrer_policy(ReferrerPolicy::from_string(attribute(HTML::AttributeNames::referrerpolicy)));
+
+        // FIXME: 21. Set request's priority to the current state of the element's fetchpriority attribute.
+
+        // FIXME: 22. Let delay load event be true if the img's lazy loading attribute is in the Eager state, or if scripting is disabled for the img, and false otherwise.
+
+        // FIXME: 23. If the will lazy load element steps given the img return true, then:
+        // FIXME:     1. Set the img's lazy load resumption steps to the rest of this algorithm starting with the step labeled fetch the image.
+        // FIXME:     2. Start intersection-observing a lazy loading element for the img element.
+        // FIXME:     3. Return.
+
+        Fetch::Infrastructure::FetchAlgorithms::Input fetch_algorithms_input {};
+        fetch_algorithms_input.process_response = [this, image_request, url_string](JS::NonnullGCPtr<Fetch::Infrastructure::Response> response) {
+            // 25. As soon as possible, jump to the first applicable entry from the following list:
+
+            // FIXME: - If the resource type is multipart/x-mixed-replace
+
+            // - If the resource type and data corresponds to a supported image format, as described below
+            // - The next task that is queued by the networking task source while the image is being fetched must run the following steps:
+            queue_an_element_task(HTML::Task::Source::Networking, [this, response, image_request, url_string] {
+                // FIXME: See HTMLLinkElement::default_fetch_and_process_linked_resource for thorough notes on the workaround
+                //        added here for CORS cross-origin responses. The gist is that all cross-origin responses will have a
+                //        null bodyBytes. So we must read the actual body from the unsafe response.
+                //        https://github.com/whatwg/html/issues/9066
+                if (response->is_cors_cross_origin() && !response->body().has_value() && response->unsafe_response()->body().has_value()) {
+                    auto unsafe_response = static_cast<Fetch::Infrastructure::OpaqueFilteredResponse const&>(*response).internal_response();
+                    unsafe_response->body()->fully_read(
+                                               realm(),
+                                               [image_request, url_string, this](ByteBuffer data) {
+                                                   handle_successful_fetch(url_string, image_request, move(data));
+                                               },
+                                               [this](auto&) {
+                                                   handle_failed_fetch();
+                                               },
```

Comment:
I blame clang-format for whatever this formatting is, but since these 2 lambdas are the same here and in the happy path below, can we create them once ahead of time and move them into these calls? Just to make it a bit cleaner to read, we do that elsewhere as well. Also you don't need to static_cast here, we can just call `unsafe_response()` on the `response` to get its internal response (that method didn't exist at the time that `<link>` added this workaround):

```c++
auto process_body = [this, image_request, url_string](ByteBuffer data) {
    handle_successful_fetch(url_string, image_request, move(data));
};
auto process_body_error = [this](auto&) {
    handle_failed_fetch();
};

if (response->is_cors_cross_origin() && !response->body().has_value() && response->unsafe_response()->body().has_value()) {
    response->unsafe_response()->body()->fully_read(realm, move(process_body), move(process_body_error), JS::NonnullGCPtr { realm.global_object() }).release_value_but_fixme_should_propagate_errors();
} else {
    response->body()->fully_read(realm, move(process_body), move(process_body_error), JS::NonnullGCPtr { realm.global_object() }).release_value_but_fixme_should_propagate_errors();
}
```